### ISSUE TITLE
should chpldoc show cnames for extern/export procs?

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#10387]
E.g. 

``` chapel
  export "chapel_addone" proc addone(arg:c_int):c_int {
    return arg+1;
  }
```

How should chpldoc show this?

It currently ignores the cname. Note that the cname could be a param function call, and then chpldoc (presumably) couldn't show its value.

**Comment by bradcray** [comment]
I think for the string literal case, it should show it pretty similar to how it looks in the code.  The point about param string names is a good one, though and might (ultimately) warrant having chpldoc run later in the compiler?

**Comment by mppf** [comment]
Maybe, or it could just show the expression that generates the C name.

But either way, the C name isn't for people using the *Chapel code from Chapel* and so I'd lean away from including it in chpldocs.

**Comment by lydia-duncan** [comment]
> But either way, the C name isn't for people using the Chapel code from Chapel and so I'd lean away from including it in chpldocs.

On the one hand, it is something that is explicitly stated about the function and so should be documented for clients of it.  On the other hand, we don't generate documentation information about what the generated C code would look like normally (and I don't think we should).

**Comment by bradcray** [comment]
I was imagining that I was using some module that exported some functions and that, as such, I'd be using the code it generated as well as a client of the resulting library / program.  Without knowing the exported name, how would I know what to invoke from my C program?  (I guess some answers might be: "look at the Chapel source code and reason about the `export` clause" or "look at the generated header file", but the former seems clunky and the latter may not help at all if it uses a dramatically different name.

If I'm in the minority, though, I'm fine with not creating work for ourselves.

**Comment by mppf** [comment]
I think a module meant for use from C just needs documentation focused on doing that and that the `chpldoc` output is different from what the "here's how you call it from C" documentation should look like that we should just view it as a separate piece of documentation.

In particular, I'd expect that the "how to use it from C" documentation might be in the grand top-level module doc string, and that the individual function's chpldoc won't be that useful in that context.

But in any case aren't we only talking about `export` functions right now? Isn't there even less rationale to show the param cname for an `extern` function in the chpldoc?

**Comment by lydia-duncan** [comment]
> But in any case aren't we only talking about export functions right now? Isn't there even less rationale to show the param cname for an extern function in the chpldoc?

Yeah, I think if we did want this information included, we would only want it for `export`ed functions.  It seems like an implementation detail for extern functions (and as such something that should be hidden)

**Comment by bradcray** [comment]
I agree with Lydia and was focusing solely on the `export` case.  I don't think the string is important for `extern` declarations.
